crates/context/src/lib.rs

```diff
@@ -307,11 +307,18 @@ impl ContextManager {
         blob_id: calimero_primitives::blobs::BlobId,
         source: calimero_primitives::application::ApplicationSource,
         version: Option<semver::Version>,
+        contract_app_id: Option<String>,
     ) -> eyre::Result<calimero_primitives::application::ApplicationId> {
+        let contract_app_id: Box<str> = match contract_app_id {
+            Some(s) => s.into_boxed_str(),
+            None => Box::from(""),
+        };
+
         let application = calimero_store::types::ApplicationMeta {
             blob: calimero_store::key::BlobMeta::new(blob_id),
             version: version.map(|v| v.to_string().into_boxed_str()),
             source: source.to_string().into_boxed_str(),
+            contract_app_id: Some(contract_app_id),
```

Comment:
hm if value is None and type is Option, why not leave it None?

Reply:
ApplicationMeta is used somewhere else where contract_app_id can be None, in this case its always Some as contract_app_id is transformed into Box<str> either app_id value or "".

Reply:
I followed what was already implemented there and using Box probably has its reason. cc @miraclx Could you please answer this ?

Reply:
As mentioned above, this will change. But aside from that, will it ever be `None`?